### DISCUSSION TITLE
Prefetch is not called when navigating to the same route with a different param

### DISCUSSION THREAD

**Discussion by lazaroofarrill** [quasarframework/quasar#10343] [General - Components / Directives / etc] [answered]
I have this route

```
path: 'news', component: () => import('pages/News.vue'), meta: {
          nav: newsLinks,
          translations: 'news.links',
          drawerInjection: 'news-sidebar'
        },
        children: [
          {
            path: 'detail/:id', component: () => import('pages/news/Details.vue'), props: true
          },
...
}
```

When navigating to `/news/detail/:id` I prefetch the news with the given id. The problem is when I try to navigate inside this page with a link to another news, say ,`/news/detail/2` to `/news/detail/3`. In this case prefetch is not called. How should I go about this.

**Comment by ontwikkelfabriek** [comment, marked as answer]
@lazaroofarrill 

From the Vue docs:
"_You can either watch the $route object to react to those changes, or use the beforeRouteUpdate in-component guard_"
https://router.vuejs.org/guide/advanced/navigation-guards.html#global-before-guards

I would use `beforeRouterUpdate` (In-Component guard):
https://router.vuejs.org/guide/advanced/navigation-guards.html#in-component-guards

**Reply by lazaroofarrill** [reply]
Thank you very much

**Reply by gh81997167** [reply]
> 
I have the same problem, can you show me how to use beforeRouteUpdate with the Prefetch hook

**Reply by lazaroofarrill** [reply]
```ts
 preFetch({ store, currentRoute }) {
    return store.dispatch('api/fetchNewsDetails', currentRoute.params.id)
  },
  async beforeRouteUpdate(to, from, next) {
    await this.$store.dispatch('api/fetchNewsDetails', to.params.id)

    next()
  },
```

**Reply by TobyMosque** [reply]
> > 
> 
> I have the same problem, can you show me how to use beforeRouteUpdate with the Prefetch hook

do you can open a issue?
https://quasar.dev/quasar-cli-webpack/prefetch-feature#when-prefetch-gets-activated

| Route being visited | Hooks called from | Observations |
|---|---|---|
| Route being visited | Hooks called from | Observations |
| /shop/product/pyjamas | ShopProduct | - |
| /shop/product/shoes | ShopProduct | Quasar notices the same component is already rendered, but the route has been updated and it has route params, so it calls the hook again. |
